InterfaceMATLAB/tsgCoreTests.m

```diff
@@ -445,6 +445,44 @@ function tsgCoreTests()
 end
 tsgDeleteGrid(lGrid);
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%%                     tsgMakeFourier()                             %%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% test transform
+[lGrid, p] = tsgMakeFourier('_tsgcoretests_lgrid', 3, 1, 'level', 1, [3.0 5.0; -7.0 -6.0; -12.0 17.0]);
+if ((abs(max(p(:, 1)) - 13.0/3.0) > 1.E-11) || (abs(min(p(:, 1)) - 3.0) > 1.E-11) ...
+    || (abs(max(p(:, 2)) + 19.0/3.0) > 1.E-11) || (abs(min(p(:, 2)) + 7.0) > 1.E-11) ...
+    || (abs(max(p(:, 3)) - 22.0/3.0) > 1.E-11) || (abs(min(p(:, 3)) + 12.0) > 1.E-11))
+    error('Mismatch in tsgMakeFourier: transform');
+end
+[w, p] = tsgGetQuadrature(lGrid);
+if ((abs(norm(sum(w)) - 58.0) > 1.E-11) || (abs(max(p(:, 1)) - 13.0/3.0) > 1.E-11) ...
+    || (abs(min(p(:, 1)) - 3.0) > 1.E-11) || (abs(max(p(:, 2)) + 19.0/3.0) > 1.E-11) ...
+    || (abs(min(p(:, 2)) + 7.0) > 1.E-11) || (abs(max(p(:, 3)) - 22.0/3.0) > 1.E-11) ...
+    || (abs(min(p(:, 3)) + 12.0) > 1.E-11))
+    error('Mismatch in tsgMakeFourier: getQuadrature');
+end
+
+% correctness of 1-D
+[lGrid, pf] = tsgMakeFourier('_tsgcoretests_lgrid', 1, 1, 'level', 2);
+bmark = [0.0; 1.0/3.0; 2.0/3.0; 1.0/9.0; 2.0/9.0; 4.0/9.0; 5.0/9.0; 7.0/9.0; 8.0/9.0];
+if (norm(pf - bmark) > 1.E-11)
+    error('Mismatch in tsgMakeFourier: points');
+end
+
+% level limits
+[lGrid, p] = tsgMakeFourier('_tsgcoretests_lgrid', 3, 1, 'level', 2, [], [], [], [], [0, 1, 2]);
+if (min(abs(p(:,1))) > 1.E-8)
+    error('Mismatch in tsgMakeFourier: level limit, dim 1');
+end
+if (min(abs(p(:,2) - 2.0/3.0)) > 1.E-8)
```

Comment:
* this will be false if `2.0/3.0` is missing from the grid, the point of level limits is to ensure that there are no points from an upper level, i.e., you should look if `1.0/9.0` is included by mistake.